src/object/schedcontrol.c

```diff
@@ -115,23 +117,23 @@ static exception_t decodeSchedControl_Configure(word_t length, cap_t cap, extra_
         return EXCEPTION_SYSCALL_ERROR;
     }
 
-    if (budget_us > getMaxUsToTicks() || budget_us < MIN_BUDGET_US) {
+    if (budget > getMaxUsToTicks() || budget < MIN_BUDGET) {
         userError("SchedControl_Configure: budget out of range.");
         current_syscall_error.type = seL4_RangeError;
         current_syscall_error.rangeErrorMin = MIN_BUDGET_US;
-        current_syscall_error.rangeErrorMax = getMaxUsToTicks();
+        current_syscall_error.rangeErrorMax = getMaxTicksToUs();
         return EXCEPTION_SYSCALL_ERROR;
     }
 
-    if (period_us > getMaxUsToTicks() || period_us < MIN_BUDGET_US) {
+    if (period > getMaxUsToTicks() || period < MIN_BUDGET) {
         userError("SchedControl_Configure: period out of range.");
         current_syscall_error.type = seL4_RangeError;
         current_syscall_error.rangeErrorMin = MIN_BUDGET_US;
-        current_syscall_error.rangeErrorMax = getMaxUsToTicks();
+        current_syscall_error.rangeErrorMax = getMaxTicksToUs();
```

Comment:
Since the user passed in US, the range should be expressed in US.

Reply:
Yes that is what this change is ensuring isn't it?

Reply:
Oh, I seem to have confused the actual purpose of `getMaxUsToTicks` and `getMaxTicksToUs`.

Reply:
As far as I recall, `getMaxUsToTicks` should be returning the max value that can be passed to `usToTicks`.